enumerables.rb

```diff
@@ -0,0 +1,188 @@
+    return sub_param.match?(value) if sub_param.class == Regexp
+    # when a class is passed as an argument
+    return value.class == sub_param if sub_param.is_a?(Class)
+    # when sub_param is given
+    return sub_param == value unless sub_param.nil?
+
+    yield(value) ? true : false
+  end
+  # rubocop:enable Metrics/PerceivedComplexity,Metrics/CyclomaticComplexity
+
+  # Refactoring my_all?, enlightened by mentor Rory Heiller
+  def my_all?(main_param = nil)
+    result = true
+    my_each do |element| # iterate over self
+      condition = my_all_any_none_helper(main_param, element) { |i| block_given? ? yield(i) : i }
+      result &&= condition # i equals each element here
+      break if result == false # once find a false, exit the loop
+    end
+    result
+  end
+
+  # 5. Create my_any?
+  def my_any?(main_param = nil)
+    result = false
+    my_each do |element| # iterate over self
+      condition = my_all_any_none_helper(main_param, element) { |i| block_given? ? yield(i) : i }
+      result ||= condition
+      break if result == true # once got a true, exit the loop
+    end
+    result
+  end
+
+  # 6. Create my_none?
+  def my_none?(main_param = nil)
+    result = false
+    my_each do |element| # iterate over self
+      condition = my_all_any_none_helper(main_param, element) { |i| block_given? ? yield(i) : i }
+      result ||= condition
+      break if result == true # once got a true, exit the loop
+    end
+    !result # inverse of my_any?
+  end
+
+  # 7. Create my_count
+  def my_count(param = nil)
+    counter = 0
+    return length if param.nil? && !block_given?
+
+    my_each do |i|
+      condition = yield(i) if block_given?
+      counter += 1 if condition || param == i
+    end
+    counter
+  end
+
+  # 8. Create my_map
+  #   def my_map
+  #     arr = []
+  #     each do |i|
+  #       condition = yield(i)
+  #       arr << condition
+  #     end
+  #     arr
+  #   end
+
+  # 9. Create my_inject
+  def my_inject(initial = nil, sym = nil)
+    if block_given?
+      if !initial.nil? # Check if arguments are passed in
+        result = initial
+        my_each { |element| result = yield(result, element) }
+      else
+        result = self[0] # Result = first element if no argument
+        self[1..-1].my_each { |element| result = yield(result, element) }
+      end
+    elsif !sym.nil? # if Symbol is given as 2nd parameter
+      result = initial
+      my_each { |element| result = result.send(sym, element) } # .send invoke method by Symbol
+    elsif !initial.nil? # if initial is given only as a symbol
+      result = self[0]
+      self[1..-1].my_each { |element| result = result.send(initial, element) }
+    end
+    result
+  end
+
+  # 11. my_map (proc only)
+  #   def my_map(&my_proc)
+  #     arr = []
+  #     each do |i|
+  #       condition = my_proc.call(i)
+  #       arr << condition
+  #     end
+  #     arr
+  #   end
+
+  # 12. my_map with proc/block
+  def my_map(&my_proc)
```

Comment:
* [ ] should have working `my_map`  identical and returns the same thing as ruby's [`map`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-map). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_map.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
```p [1,3].map``` ```!=```  ```p [1,3].my_map```